frameworks/kafka/setup-helper/setup_helper.go

```diff
@@ -162,14 +163,26 @@ func setListeners() error {
 	} else { // No TLS, no Kerberos, Plaintext only
 		listeners = append(listeners,
 			getListener("PLAINTEXT", brokerPort))
+		// NOTE: To be consistent with the legacy behavior of the 2.0.X Kafka series,
+		// we advertise the IP address rather than the host name.
 		advertisedListeners = append(advertisedListeners,
-			getAdvertisedListener("PLAINTEXT", brokerPort))
+			getListener("PLAINTEXT", brokerPort))
 	}
 
 	err := writeToWorkingDirectory(listenersProperty,
 		"listeners="+strings.Join(listeners, ","))
-	err = writeToWorkingDirectory(advertisedListenersProperty,
-		"advertised.listeners="+strings.Join(advertisedListeners, ","))
+
+	// NOTE: To be consistent with the legacy behavior of the 2.0.X Kafka series,
+	// when there is no security enabled, we must honor the kafka.kafka_advertise_host_ip
+	// configuration parameter
+	if !kerberosEnabled && !tlsEncryptionEnabled && !getBooleanEnvvar(advertiseHostIPEnvvar) {
```

Comment:
How about `a || b || c` and swap the execution blocks?

(i.e. `(!a && !b && !c) == !(a || b || c)` - my boolean algebra may be a bit rusty though)

Reply:
si